### PR TITLE
add tests for convertToMap, splitEqual functions in istioctl

### PR DESCRIPTION
**Pull request by hacktron95**
Please provide a description for what this PR is for:

increase code coverage by adding more tests in istioctl

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[x] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: istioctl/cmd/add-to-mesh_test.go [hunk 1]
```diff
@@ -286,3 +286,48 @@ func mockDynamicClientGenerator(dynamicConfigs []runtime.Object) func(kubeconfig
 	}
 	return outFactory
 }
+
+func Test_splitEqual(t *testing.T) {
```
Comment (adiprerepa) [inline istioctl/cmd/add-to-mesh_test.go:290]: should be `TestSplitEqual`

File: istioctl/cmd/add-to-mesh_test.go [hunk 2]
```diff
@@ -286,3 +286,48 @@ func mockDynamicClientGenerator(dynamicConfigs []runtime.Object) func(kubeconfig
...
+				t.Errorf("splitEqual(%v) got = %v, want %v", tt.arg, gotKey, tt.wantKey)
+			}
+			if gotValue != tt.wantValue {
+				t.Errorf("splitEqual(%v) got1 = %v, want %v", tt.arg, gotValue, tt.wantValue)
+			}
+		})
+	}
+}
+
+func Test_convertToMap(t *testing.T) {
```
Comment (adiprerepa) [inline istioctl/cmd/add-to-mesh_test.go:315]: should be `TestConvertToMap`

**Comment by esnible** [comment]
The tests look reasonable.  There is no Issue.  Did you have a problem that inspired you to write these tests?

**Comment by hacktron95** [comment]
@esnible I submitted this as my first PR,
I'm trying to get familiar with istioctl (so I can contribute more) and found that the test coverage is around 49%, and thought I can start contributing by writing more tests (as it will help me understand the code-base).

**Comment by esnible** [comment]
/retest
Not sure why compilation failed for the lint and the test.

**Comment by adiprerepa** [comment]
@hacktron95 try rebasing. Maybe the function was removed.

**Comment by hacktron95** [comment]
> @hacktron95 try rebasing. Maybe the function was removed.

I did rebase but no changes came up, also I can run `make test`, `make lint-go`, `make format-go` with no issues or errors on my local machine, I also can build `make build`.

**Comment by shamsher31** [comment]
> I did rebase but no changes came up, also I can run make test, make lint-go, make format-go with no issues or errors on my local machine, I also can build make build.

Most of the time you only need to run the following 3 commands.

1. `make lint`
2. `make gen` (sometimes `make refresh-goldens` [when YAML files are changed](https://github.com/istio/istio/blob/master/manifests/charts/UPDATING-CHARTS.md#step-5-update-golden-files)) 
3. `make build`

**Comment by shamsher31** [comment]
@hacktron95 I see the following lint error.

```
istioctl/cmd/add-to-mesh_test.go:329:14: undefined: convertToMap
```

Even though the test and the func is in the same package, the `istioctl` is not able to find it. Can you export `convertToMap` (write `ConvertToMap`) and try running [above](https://github.com/istio/istio/pull/26908#issuecomment-686372873) commands.

File: istioctl/cmd/add-to-mesh_test.go [hunk 3]
```diff
@@ -286,3 +287,48 @@ func mockDynamicClientGenerator(dynamicConfigs []runtime.Object) func(kubeconfig
...
+		want map[string]string
+	}{
+		{name: "empty", arg: []string{""}, want: map[string]string{"": ""}},
+		{name: "one-valid", arg: []string{"key=value"}, want: map[string]string{"key": "value"}},
+		{name: "one-valid-double-equals", arg: []string{"key==value"}, want: map[string]string{"key": "=value"}},
+		{name: "one-key-only", arg: []string{"key"}, want: map[string]string{"key": ""}},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := convertToMap(tt.arg); !reflect.DeepEqual(got, tt.want) {
```
Comment (shamsher31) [inline istioctl/cmd/add-to-mesh_test.go:329]: Export the original func and try. See this https://github.com/istio/istio/pull/26908#issuecomment-686375465

**Comment by hacktron95** [comment]
@shamsher31 It took me a while to understand what was the issue, the master branch had a function name change totally, that's why the test wasn't able to identify the function.

**Review by shamsher31 — APPROVED** [review]
LGTM.

**Pull request merged**